packages/myst-cli/src/init/jupyter-book/syntax.ts

```diff
@@ -0,0 +1,279 @@
+    let indentSize = 0;
+
+    const entries: LegacyGlossaryItem[] = [];
+
+    // Parse lines into separate entries
+    for (let i = 0; i < nodeLines.length; i++) {
+      const line = nodeLines[i];
+      // Is the line a comment?
+      if (/^\.\.\s/.test(line) || !line.length) {
+        continue;
+      }
+      // Is the line a non-whitespace-leading line (term declaration)?
+      else if (/^[^\s]/.test(line[0])) {
+        // Comment
+        if (line.startsWith('.. ')) {
+          continue;
+        }
+
+        // Do we need to create a new entry?
+        if (inDefinition || !entries.length) {
+          // Close the current definition, open a new term
+          entries.push({
+            definitionLines: [],
+            termLines: [{ content: line, offset: i }],
+          });
+          inDefinition = false;
+        }
+        // Can we extend existing entry with an additional term?
+        else if (entries.length) {
+          entries[entries.length - 1].termLines.push({ content: line, offset: i });
+        }
+      }
+      // Open a definition
+      else {
+        inDefinition = true;
+        indentSize = line.length - line.replace(/^\s+/, '').length;
+
+        if (entries.length) {
+          entries[entries.length - 1].definitionLines.push({
+            content: line.slice(indentSize),
+            offset: i,
+          });
+        }
+      }
+    }
+
+    // Build glossary
+    const newLines: string[] = [];
+
+    for (let i = 0; i < entries.length; i++) {
+      const entry = entries[i];
+      const { termLines, definitionLines } = entry;
+
+      const [firstDefinitionLine, ...restDefinitionLines] = definitionLines;
+      const [firstTerm, ...restTerms] = termLines;
+
+      // Initial definition
+      const firstTermValue = firstTerm.content.split(/\s+:\s+/, 1)[0];
+      newLines.push(
+        firstTermValue,
+        `: ${firstDefinitionLine.content}`,
+        ...restDefinitionLines.map((line) => `  ${line.content}`),
+      );
+      if (restTerms) {
+        // Terms can contain markup, but we need the text-form to create a term reference
+        // TODO: what if something magical like an xref is used here? Assume not.
+        const parsedTerm = mystParse(firstTermValue);
+        const termName = toText(parsedTerm);
+        for (const { content } of restTerms) {
+          const term = content.split(/\s+:\s+/, 1)[0];
+          newLines.push(
+            // Separate from parent term
+            '',
+            term,
+            `: {term}\`${termName}\``,
+          );
+        }
+      }
+
+      // Will there be following terms?
+      const isFinalEntry = i === entries.length - 1;
+      if (!isFinalEntry) {
+        newLines.push('');
+      }
+    }
+    const nodeSpan = { start: node.position?.start?.line, stop: node.position?.end?.line };
+    const spanLength = nodeSpan.stop! - nodeSpan.start! - 1;
+    documentLines.splice(nodeSpan.start! + editOffset, spanLength, ...newLines);
```

Comment:
Are these `!` assertions reasonable?

Reply:
Similar to the note above, I would bail if there are not positions. Not sure if that is possible in this case though. Given that there are unit tests now, this should be safe!